cli/connhelper/connhelper.go

```diff
@@ -79,3 +80,14 @@ func addSSHTimeout(sshFlags []string) []string {
 	}
 	return sshFlags
 }
+
+// disablePseudoTerminalAllocation disables pseudo-terminal allocation to
+// prevent SSH from executing as a login shell
+func disablePseudoTerminalAllocation(sshFlags []string) []string {
+	for _, flag := range sshFlags {
+		if flag == "-T" {
+			return sshFlags
+		}
+	}
+	return append(sshFlags, "-T")
+}
```

Comment:
Shouldn't this also check if `-t` has been specified explicitly and don't apply the `-T` in this case?

Reply:
Not sure about that – this PR disables pseudo-tty allocation for `ssh`. Using `-t` should continue working regardless, as that impacts allocating a pseudo-tty inside the container, but I don't think that somehow requires `ssh` to be a pseudo-tty as well.

Reply:
I guess this should be easy to check – I'll take a look.

Reply:
This works fine/as expected.

Reply:
Ah, yes, we should probably skip if that's the case.

Probably not immediately critical for us because we don't set other options, but other users of the module technically could. And it doesn't seem like it would hurt to add the check.

Reply:
Ahhh 😅 for sure. I think that's okay, do we expect users to set flags in their context?

Reply:
Ah right, you can't set that via context 😅 It's only with this `GetConnectionHelperWithSSHOpts` that isn't used on our side.

Reply:
I don't think there's ways to customise these currently, unless there's some options inherited from the ssh-config. The ability to set options was added in https://github.com/docker/cli/pull/2541#discussion_r480098871 (some of the discussion / context related to it in that comment)